### DISCUSSION TITLE
How I can generate user inputs choice based on files in the repo?

### DISCUSSION THREAD

**Discussion by vitalykarasik** [community/community#25291] [Actions]
<p>How I can generate user inputs choice based on files in the repo? - A la “active choices” in Jenkins.<br>
I.e. if I have “one.xml, two.xml, three.xml” into my repo, I’d like to have them into “choice:” list.</p>
<p>Thanks, Vitaly</p>

**Comment by jfullstackdev** [comment]
<p>you mean, just like this ?</p>
<aside class="onebox allowlistedgeneric" data-onebox-src="https://www.w3schools.com/howto/howto_js_dropdown.asp">
  <header class="source">
      <img src="https://user-images.githubusercontent.com/47092464/181090102-74bef358-1cd5-424b-a7aa-5ddaa8fc07a8.png" class="site-icon" width="48" height="48">

      <a href="https://www.w3schools.com/howto/howto_js_dropdown.asp" target="_blank" rel="noopener">w3schools.com</a>
  </header>

  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:436/228;"><img src="https://user-images.githubusercontent.com/47092464/181090103-b0512dbf-fd21-4de5-b6fb-edbb06fcfe50.png" class="thumbnail" width="436" height="228"></div>

<h3><a href="https://www.w3schools.com/howto/howto_js_dropdown.asp" target="_blank" rel="noopener">How To Create a Dropdown Menu With CSS and JavaScript</a></h3>

  <p>W3Schools offers free online tutorials, references and exercises in all the major languages of the web. Covering popular subjects like HTML, CSS, JavaScript, Python, SQL, Java, and many, many more.</p>


  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

**Comment by vitalykarasik** [comment]
<p>Thank you, but not exactly.<br>
There is “type: choice” input type for manually triggered workflows. (<a href="https://github.blog/changelog/2021-11-10-github-actions-input-types-for-manual-workflows/" class="inline-onebox" rel="noopener nofollow ugc">GitHub Actions: Input types for manual workflows | GitHub Changelog</a>).<br>
I’d like to populate this list dynamically with a files list in my repo.</p>

**Comment by Simran-B** [comment, marked as answer]
<p>There’s no support for a file picker, and neither for dynamic options in general, which was also requested here: <a href="https://github.community/t/can-workflow-dispatch-input-be-option-list/127338/59" class="inline-onebox">Can workflow_dispatch input be option list? - #59 by brcarp</a></p>
